pkg/collector/upgrade/upgrade.go

```diff
@@ -22,34 +22,44 @@ import (
 
 	semver "github.com/Masterminds/semver/v3"
 	"github.com/go-logr/logr"
+	"k8s.io/client-go/tools/record"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
 	"github.com/open-telemetry/opentelemetry-operator/internal/version"
 )
 
+type VersionUpgrade struct {
+	Log      logr.Logger
```

Comment:
iirc `logger` is more used in this codebase

Reply:
Sure, I'm using it in different upgrade routines.